File: frontend/src/services/dashboardService.js
import apiClient from './apiClient';

const dashboardService = {
  async getAdminDashboard() {
    const { data } = await apiClient.get('/dashboard/admin');
    return data.data;
  },
  async getDoctorDashboard(doctorId) {
    const { data } = await apiClient.get(`/dashboard/doctor/${doctorId}`);
    return data.data;
  },
};

export default dashboardService;

File: backend/src/main/java/com/hospital/hms/service/DashboardService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.response.AdminDashboardResponse;
import com.hospital.hms.dto.response.DoctorDashboardResponse;

public interface DashboardService {

    AdminDashboardResponse getAdminDashboard();

    DoctorDashboardResponse getDoctorDashboard(Long doctorId);
}
